src/utils.ts

```diff
@@ -31,10 +32,57 @@ export const doCommandFromClient = async function doCommandFromClient(
 
   options.requestLogger?.(request);
 
-  const response = await doCommander(request);
+  const response = await doCommander(request, callOptions);
   const result = response.result?.toJson();
   if (!result) {
     return {};
   }
   return result;
 };
+
+export const enableDebugLogging = (
+  key?: string,
+  opts?: CallOptions
+): CallOptions => {
+  const finalOpts = opts ?? { headers: {} as Record<string, string> };
+  let finalKey = '';
+  if (key) {
+    finalKey = key;
+  } else {
+    const letters = 'abcdefghijklmnopqrstuvwxyz';
+    for (let i = 0; i < 6; i += 1) {
+      finalKey += letters[Math.floor(Math.random() * 26)];
+    }
+  }
+  (finalOpts.headers as Record<string, string>).dtname = finalKey;
+  return finalOpts;
+};
+
+export const disableDebugLogging = (opts: CallOptions): void => {
+  if (opts.headers) {
+    const { dtname, ...remainingHeaders } = opts.headers as Record<
+      string,
+      string
+    >;
+    opts.headers = remainingHeaders;
+  }
+};
+
+export const addMetadata = (
+  key: string,
+  value: string,
+  opts?: CallOptions
+): CallOptions => {
+  const finalOpts =
+    opts ?? ({ headers: {} as Record<string, string> } as CallOptions);
+  (finalOpts.headers as Record<string, string>)[key] = value;
+  return finalOpts;
+};
+
+export const deleteMetadata = (opts: CallOptions, key: string): void => {
+  const { [key]: _, ...remainingHeaders } = opts.headers as Record<
+    string,
+    string
+  >;
+  opts.headers = remainingHeaders;
+};
```

Comment:
Are these helpers used? Or just for potential users ?

Reply:
These two helpers not being used right now, but I thought it would be good to include them for potential usage. This was also introduced in the Python SDK, so I thought it would be a good idea to keep things similar. Should I remove it?

Reply:
I think we should leave it in personally. We're now offering users access to passing metadata so we might as well make it easy for them to do so.

Reply:
we can leave in. i was just wondering if i missed something